osf/metrics.py

```diff
@@ -175,3 +175,99 @@ class PreprintView(BasePreprintMetric):
 
 class PreprintDownload(BasePreprintMetric):
     pass
+
+
+class UserInstitutionProjectCounts(MetricMixin, metrics.Metric):
+    user_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    institution_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    # TODO: Uncomment following line and include department in metrics when available as a OSFUser attribute
+    # department = metrics.Keyword(index=True, doc_values=True, required=False)
+    public_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+    private_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+
+    class Index:
+        settings = {
+            'number_of_shards': 1,
+            'number_of_replicas': 1,
+            'refresh_interval': '1s',
+        }
+
+    @classmethod
+    def record_user_institution_project_counts(cls, user, institution, public_project_count, private_project_count, **kwargs):
+        return cls.record(
+            user_id=user._id,
+            institution_id=institution._id,
+            # TODO: Uncomment following line and include department in metrics when available as a OSFUser attribute
+            # department=user.department,
+            public_project_count=public_project_count,
+            private_project_count=private_project_count,
+            **kwargs
+        )
+
+    @classmethod
+    def get_latest_user_institution_project_counts(cls, user, institution):
```

Comment:
It isn't an explicit requirement - I added them to complement the record methods and have more complete metrics classes. Good call with returning the document instead of the tuple - I'll change that.